pipeline/authn/authenticator_oauth2_introspection_cache_test.go

```diff
@@ -0,0 +1,91 @@
+package authn
+
+import (
+	"testing"
+	"time"
+
+	"github.com/ory/fosite"
+	"github.com/ory/oathkeeper/driver/configuration"
+	"github.com/ory/viper"
+	"github.com/ory/x/logrusx"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCache(t *testing.T) {
+	viper.Reset()
+
+	viper.Set("authenticators.oauth2_introspection.config.cache.enabled", true)
+	viper.Set("authenticators.oauth2_introspection.config.introspection_url", "http://localhost:8080/")
+
+	logger := logrusx.New("", "")
+	c := configuration.NewViperProvider(logger)
+	a := NewAuthenticatorOAuth2Introspection(c, logger)
+	assert.Equal(t, "oauth2_introspection", a.GetID())
+
+	config, _, err := a.Config(nil)
+	require.NoError(t, err)
+
+	t.Run("method=tokenToCache", func(t *testing.T) {
+		t.Run("case=cache value", func(t *testing.T) {
+			i := &AuthenticatorOAuth2IntrospectionResult{
+				Active: true,
+				Extra:  map[string]interface{}{"extra": "foo"},
+			}
+
+			a.tokenToCache(config, i, "token", fosite.WildcardScopeStrategy)
+			// wait cache to save value
+			time.Sleep(time.Millisecond * 10)
```

Comment:
If you update ristretto to the newest version it has a new method called wait which ensures that the cache worker has completed it’s work. It is more reliable and less flaky than waiting here! Could you maybe use that instead? :)

Reply:
update ristretto to v0.1.0 will cause an issue: flag redefined: v.
which defined in :

```go
# github.com/google/martian/init.go
var (
	level = flag.Int("v", 0, "log level")
)
```

```go
# github.com/google/golang/glog.go
func init() {
	flag.BoolVar(&logging.toStderr, "logtostderr", false, "log to standard error instead of files")
	flag.BoolVar(&logging.alsoToStderr, "alsologtostderr", false, "log to standard error as well as files")
	flag.Var(&logging.verbosity, "v", "log level for V logs")
	flag.Var(&logging.stderrThreshold, "stderrthreshold", "logs at or above this threshold go to stderr")
	flag.Var(&logging.vmodule, "vmodule", "comma-separated list of pattern=N settings for file-filtered logging")
	flag.Var(&logging.traceLocation, "log_backtrace_at", "when logging hits line file:N, emit a stack trace")

	// Default stderrThreshold is ERROR.
	logging.stderrThreshold = errorLog

	logging.setVState(0, nil, false)
	go logging.flushDaemon()
}
```

Reply:
🤦‍♂️ ok then keeping it as is is fine! Thank you :)